src/Aspire.Hosting/Dcp/DcpDependencyCheck.cs

```diff
@@ -0,0 +1,218 @@
+
+    [GeneratedRegex("[^\\d\\.].*$")]
+    private static partial Regex VersionRegex();
+
+    private readonly DistributedApplicationModel _applicationModel;
+    private readonly DcpOptions _dcpOptions;
+
+    public DcpDependencyCheck(
+        DistributedApplicationModel applicationModel,
+        IOptions<DcpOptions> dcpOptions)
+    {
+        _applicationModel = applicationModel;
+        _dcpOptions = dcpOptions.Value;
+    }
+
+    public async Task EnsureDcpDependenciesAsync(CancellationToken cancellationToken = default)
+    {
+        string? dcpPath = _dcpOptions.CliPath;
+        string? containerRuntime = _dcpOptions.ContainerRuntime;
+
+        if (!File.Exists(dcpPath))
+        {
+            throw new FileNotFoundException($"The Aspire orchestration component is not installed at \"{dcpPath}\". The application cannot be run without it.", dcpPath);
+        }
+
+        IAsyncDisposable? processDisposable = null;
+        Task<ProcessResult> task;
+
+        try
+        {
+            var outputStringBuilder = new StringBuilder();
+
+            var arguments = "info";
+            if (!string.IsNullOrEmpty(containerRuntime))
+            {
+                arguments += $" --container-runtime {containerRuntime}";
+            }
+
+            // Run `dcp version`
+            var processSpec = new ProcessSpec(dcpPath)
+            {
+                Arguments = arguments,
+                OnOutputData = s => outputStringBuilder.Append(s),
+            };
+
+            (task, processDisposable) = ProcessUtil.Run(processSpec);
+
+            await task.WaitAsync(TimeSpan.FromSeconds(WaitTimeForDcpInfoCommandInSeconds), cancellationToken).ConfigureAwait(false);
+
+            // Parse the output as JSON
+            var output = outputStringBuilder.ToString();
+            if (output == string.Empty)
+            {
+                return; // Best effort
+            }
+
+            var dcpInfo = JsonSerializer.Deserialize<DcpInfo>(output);
+            if (dcpInfo == null)
+            {
+                return; // Best effort
+            }
+
+            EnsureDcpVersion(dcpInfo);
+            EnsureDcpContainerRuntime(dcpInfo);
+        }
+        catch (Exception ex) when (ex is not DistributedApplicationException)
+        {
+            Console.Error.WriteLine(string.Format(
+                CultureInfo.InvariantCulture,
+                Resources.DcpDependencyCheckFailedMessage,
+                ex.ToString()
+            ));
+            Environment.Exit((int)DcpVersionCheckFailures.DcpVersionFailed);
+        }
+        finally
+        {
+            if (processDisposable != null)
+            {
+                await processDisposable.DisposeAsync().ConfigureAwait(false);
+            }
+        }
+    }
+
+    private static void EnsureDcpVersion(DcpInfo dcpInfo)
+    {
+        AspireEventSource.Instance.DcpVersionCheckStart();
+
+        try
+        {
+            string? dcpVersionString = dcpInfo.Version;
```

Comment:
```suggestion
            var dcpVersionString = dcpInfo.Version;
```
And everywhere else that's valid. Aspire uses aspnetcore's code standard, i.e. var whenever possible

Reply:
I think I got all the variable declarations I touched updated.